scripts/prepare_sector_network.py

```diff
@@ -1376,6 +1427,187 @@ def add_electricity_grid_connection(n, costs):
     ]
 
 
+def get_salt_caverns(cavern_types, fn_h2_cavern):
+    h2_caverns = pd.read_csv(fn_h2_cavern, index_col=0)
+
+    if (
+        not h2_caverns.empty
+        and options["hydrogen_underground_storage"]
+        and set(cavern_types).intersection(h2_caverns.columns)
+    ):
+        h2_caverns = h2_caverns[cavern_types].sum(axis=1)
+
+        # only use sites with at least 2 TWh potential
+        h2_caverns = h2_caverns[h2_caverns > 2]
+
+        # convert TWh to MWh
+        h2_caverns = h2_caverns * 1e6
+
+        # clip at 1000 TWh for one location
+        h2_caverns.clip(upper=1e9, inplace=True)
+
+        return h2_caverns
+    else:
+        return None
+
+
+def add_storageunits(n, costs, carriers, max_hours):
+    nodes = pop_layout.index
+
+    missing_carriers = list(set(carriers).difference(n.carriers.index))
+    n.madd("Carrier", missing_carriers)
+
```

Comment:
It would make sense to check the implemented technologies as we do in `add_stores`.

Reply:
Good point. I will add it